lib/dir_model/export.rb

```diff
@@ -1,82 +1,49 @@
 module DirModel
-  class Export
-    def path
-      generate unless generated?
-      @temp_path
-    end
+  module Export
 
-    def entry_paths
-      Dir.clean_entries(path).map {|entry| File.join path, entry }
-    end
+    attr_reader :source_model, :context
 
-    def valid?
-      !file_sources.any? { |file_source| file_source.nil? || file_source.read.nil? }
-    rescue
-      false
+    # @param [Model] source_model object to export to files
+    # @param [Hash]  context
+    def initialize(source_model, context={})
+      @source_model = source_model
+      @context      = OpenStruct.new(context)
+      @root_path    = Dir.mktmpdir
     end
 
-    def file_sources
-      self.class.files.map { |file| file_source(file) }
-    end
-    def file_source(file)
-      public_send(self.class.file_source_method_name(file))
+    def paths
+      generate unless generated?
+      entry_paths
     end
-    def file_name(file)
-      File.join(*pwd, public_send(self.class.file_name_method_name(file)))
+
+    private
+
+    def entry_paths
+      Dir.clean_entries(@root_path).map { |entry| File.join(@root_path, entry) }
     end
 
     def generated?
       !!@generated
     end
+
     def generate
       cleanup if generated?
-      @temp_path = Dir.mktmpdir
 
-      @pwd = [@temp_path]
-      _generate
+      self.class.files.each do |file_name, options|
+        dir_path  = instance_exec(&options[:path])
+        file_path = File.join(dir_path, instance_exec(&options[:name]))
```

Comment:
would be nice to be able to accept string for the options. can extract this out into two methods.